### PR TITLE
test: add e2e tests for OpenAI server

### PR DESCRIPTION
**Pull request by Vasuk12**
### Which issue(s) does this pull-request address?

Ref: #1217

### Description

This PR adds end-to-end (e2e) tests for the OpenAI chat completion server API. The tests verify the complete request/response cycle between client and server, ensuring proper handling of successful requests, authentication failures, and model not found scenarios.

### Checklist

- Added `python/tests/e2e/test_openai_chat_completion.py` with three e2e test cases:
  - `test_openai_e2e_success`: Tests successful chat completion flow with valid API key
  - `test_openai_e2e_auth_failure`: Tests authentication failure with invalid API key (401 response)
 
### Future Imporvements

The following test cases will be added to enhance coverage:

1. **Missing API key test**: Testing behavior when no `Authorization` header is provided
2. **Streaming responses test**: Testing the streaming API endpoint with `stream: true`
3. **Invalid request format test**: Test handling of malformed JSON, missing required fields, and invalid message structures
4. **Response schema validation**: Add detailed validation to ensure response structure matches OpenAI API format (checking all required fields in the response)
5. **Concurrent requests test**: Test server behavior under concurrent load with multiple simultaneous requests

#### General
- [x] I have read the appropriate contributor guide: [Python](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
/ [TypeScript](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
- [x] I have signed off on my commit: [Python instructions](https://github.com/i-am-bee/beeai-framework/blob/main/python/CONTRIBUTING.md#developer-certificate-of-origin-dco) / [TypeScript instructions](https://github.com/i-am-bee/beeai-framework/blob/main/typescript/CONTRIBUTING.md#developer-certificate-of-origin-dco)
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Appropriate label(s) added to PR: `Python` for Python changes, `TypeScript` for TypeScript changes

#### Code quality checks
- [x] Code quality checks pass: `mise check` (`mise fix` to auto-fix)

#### Testing
- [ ] Unit tests pass: `mise test:unit`
- [x] E2E tests pass: `mise test:e2e`
- [ ] Tests are included (for bug fixes or new features)

#### Documentation
- [ ] Documentation is updated
- [ ] Embedme embeds code examples in docs. To update after edits, run: Python `mise docs:fix`

### CODE REVIEW

File: python/tests/e2e/__init__.py
Comment (xjacka) [inline python/tests/e2e/__init__.py]: Please move this to the `texts/adapters/openai` folder

File: python/tests/e2e/test_openai_chat_completion.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        return self.runnable_instance
+
+
+@pytest.fixture(scope="session")
+def start_test_server() -> Generator[str, None, None]:
+    port = 19001
+
+    register()
+    with contextlib.suppress(FactoryAlreadyRegisteredError):
+        OpenAIServer.register_factory(OpenAIModel, lambda model, **kwargs: model)  # type: ignore[type-var]
```
Comment (xjacka) [inline python/tests/e2e/test_openai_chat_completion.py]: This should be removed too.

File: python/tests/e2e/test_openai_chat_completion.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    @property
+    def emitter(self) -> Emitter:
+        return Emitter.root().child(namespace=["dummy"])
+
+    @runnable_entry
+    async def run(self, input: list[AnyMessage], /, **kwargs: Unpack[RunnableOptions]) -> RunnableOutput:
+        return RunnableOutput(output=[AssistantMessage(content="pong")])
+
+
+class DummyModel(OpenAIModel):
```
Comment (xjacka) [inline python/tests/e2e/test_openai_chat_completion.py]: Please remove DummyModel. The OpenAIModel instance should be created automatically from the Runnable instance.

File: python/tests/e2e/test_openai_chat_completion.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    with contextlib.suppress(FactoryAlreadyRegisteredError):
+        OpenAIServer.register_factory(OpenAIModel, lambda model, **kwargs: model)  # type: ignore[type-var]
+
+    config = OpenAIServerConfig(
+        host="127.0.0.1",
+        port=port,
+        api_key="TEST_KEY",
+    )
+
+    server = OpenAIServer(config=config, memory_manager=UnlimitedMemoryManager())
```
Comment (xjacka) [inline python/tests/e2e/test_openai_chat_completion.py:80]: remove `memory_manager` for the completion endpoint

File: python/tests/e2e/test_openai_chat_completion.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    server.register(DummyModel())  # type: ignore[arg-type]
```
Comment (xjacka) [inline python/tests/e2e/test_openai_chat_completion.py]: You can set the name for the runnable: `server.register(DummyRunnable(), name="dummy-model")`

File: python/tests/e2e/test_openai_chat_completion.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+def test_openai_e2e_success(start_test_server: str) -> None:
+    url = f"{start_test_server}/chat/completions"
+
+    payload = {
+        "model": "dummy-model",
+        "messages": [{"role": "user", "content": "ping"}],
+        "stream": False,
+    }
+
+    headers = {"Authorization": "Bearer TEST_KEY"}
```
Comment (xjacka) [inline python/tests/e2e/test_openai_chat_completion.py]: Please extract `TEST_KEY` to a variable

**Review by xjacka — CHANGES_REQUESTED** [review]

File: python/tests/e2e/__init__.py
Comment (Vasuk12) [reply python/tests/e2e/__init__.py]: > Please move this to the `texts/adapters/openai` folder

I moved test_openai_chat_completion.py into the openai folder. Should I also proceed with deleting the entire e2e folder, including its __init__.py file?

**Comment by Vasuk12** [comment]
I've implemented the bug fix where the server now returns a 404 Not Found status code instead of a generic 500 Internal Server Error. I also added the 'test test_openai_e2e_model_not_found' to ensure this behavior is correctly validated.

Please review these changes. Let me know if there are any other related changes or edge cases you'd like me to address

File: python/tests/e2e/__init__.py
Comment (xjacka) [reply python/tests/e2e/__init__.py]: yes please

File: python/tests/adapters/openai/test_openai_chat_completion.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    data = response.json()
+    assert "detail" in data
+    assert isinstance(data["detail"], str)
+    assert "invalid" in data["detail"].lower() or "unauthorized" in data["detail"].lower()
+    assert "api key" in data["detail"].lower()
+
+
+@pytest.mark.e2e
+def test_openai_e2e_model_not_found(start_test_server: str) -> None:
+    url = f"{start_test_server}/chat/completions"
```
Comment (xjacka) [inline python/tests/adapters/openai/test_openai_chat_completion.py:146]: Can you add a responses API test too?

File: python/beeai_framework/adapters/openai/serve/chat_completion/api.py [hunk 7]
```diff
@@ -68,7 +68,16 @@ async def handler(
 
         messages = _transform_request_messages(request.messages)
 
-        runnable = self._model_factory(request.model)
+        try:
+            runnable = self._model_factory(request.model)
+        except RuntimeError as e:
```
Comment (xjacka) [inline python/beeai_framework/adapters/openai/serve/chat_completion/api.py]: You can change the error type here https://github.com/i-am-bee/beeai-framework/blob/main/python/beeai_framework/adapters/openai/serve/server.py#L80 . For example to: `raise HTTPException(status_code=404, detail=f"Model {model_id} not registered")`

**Comment by Vasuk12** [comment]
Added a responses API test, removed unused e2e folder and moved the error type to server.py.  Let me know if there are any other test cases u might need for e2e...
